beacon-chain/core/peerdas/reconstruction.go

```diff
@@ -113,18 +137,10 @@ func ReconstructDataColumnSidecars(verifiedRoSidecars []blocks.VerifiedRODataCol
 	return reconstructedVerifiedRoSidecars, nil
 }
 
-// ReconstructBlobs constructs verified read only blobs sidecars from verified read only blob sidecars.
-// The following constraints must be satisfied:
-//   - All `dataColumnSidecars` has to be committed to the same block, and
-//   - `dataColumnSidecars` must be sorted by index and should not contain duplicates.
-//   - `dataColumnSidecars` must contain either all sidecars corresponding to (non-extended) blobs,
-//     or either enough sidecars to reconstruct the blobs.
-func ReconstructBlobs(block blocks.ROBlock, verifiedDataColumnSidecars []blocks.VerifiedRODataColumn, indices []int) ([]*blocks.VerifiedROBlob, error) {
-	// Return early if no blobs are requested.
-	if len(indices) == 0 {
-		return nil, nil
-	}
-
+// validateAndPrepareDataColumns validates the input data column sidecars and returns the prepared sidecars
+// (reconstructed if necessary). This function performs common validation and reconstruction logic used by
+// both ReconstructBlobs and ReconstructBlobsData.
+func validateAndPrepareDataColumns(verifiedDataColumnSidecars []blocks.VerifiedRODataColumn) ([]blocks.VerifiedRODataColumn, error) {
```

Comment:
The name of the function does not indicate that the data columns sidecars are going to be recontructed.
==> `validateAndPrepareDataColumns` ==> `reconstructIfNeeded` ?
(We know, regarding the type of the parameters and the output, that we are talking about data column sidecars.)